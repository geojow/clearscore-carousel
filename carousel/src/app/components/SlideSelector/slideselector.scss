.slide-selector__container {
  list-style: none;
  display: flex;
  position: absolute;
  bottom: 20px;
}


.slide-selector__item:hover {
  background-color: #fff;
  height: 5px;
  width: 5px;
  margin: 0 5px;
  border-radius: 50%;
}

.slide-selector__item-active {
  background-color: #fff;
  border: solid 1px #fff;
  height: 5px;
  width: 5px;
  margin: 0 5px;
  border-radius: 50%;
}



.slide-selector__item {
  border: solid 1px #fff;
  height: 5px;
  width: 5px;
  margin: 0 5px;
  border-radius: 50%;
}


