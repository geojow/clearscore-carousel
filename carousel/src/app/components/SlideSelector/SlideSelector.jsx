import React from "react";
import PropTypes from "prop-types";
import bemHelper from "../../utils/bem";
import "./slideselector.scss";

const cn = bemHelper({ block: "slide-selector" });

const SlideSelector = ({ numberOfSlides, currentSlide, setSlide }) => (
  <ul className={cn("container")}>
    {[...Array(numberOfSlides).keys()].map((number, index) => (
      <li
        className={`${cn("item")} ${
          currentSlide === index ? `${cn("item-active")}` : ""
        }`}
        key={number}
        onClick={() => setSlide(number)}
      ></li>
    ))}
  </ul>
);

SlideSelector.propTypes = {
  numberOfSlides: PropTypes.number.isRequired,
  currentSlide: PropTypes.number.isRequired,
  setSlide: PropTypes.func.isRequired
};

export default SlideSelector;
