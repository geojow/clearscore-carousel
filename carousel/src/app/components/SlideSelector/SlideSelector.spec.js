import React from "react";
import { shallow } from "enzyme";
import SlideSelector from "./SlideSelector";

describe("SlideSelector", () => {
  let wrapper;

  beforeEach(
    () =>
      (wrapper = shallow(
        <SlideSelector
          numberOfSlides={2}
          currentSlide={0}
          setSlide={jest.fn()}
        />
      ))
  );

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render a <ul />", () => {
    expect(wrapper.find("ul").length).toEqual(1);
  });
});
