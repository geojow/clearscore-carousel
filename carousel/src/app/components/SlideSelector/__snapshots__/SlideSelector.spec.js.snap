// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideSelector should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SlideSelector
    currentSlide={0}
    numberOfSlides={2}
    setSlide={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="slide-selector__item slide-selector__item-active"
          onClick={[Function]}
        />,
        <li
          className="slide-selector__item "
          onClick={[Function]}
        />,
      ],
      "className": "slide-selector__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "className": "slide-selector__item slide-selector__item-active",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "className": "slide-selector__item ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="slide-selector__item slide-selector__item-active"
            onClick={[Function]}
          />,
          <li
            className="slide-selector__item "
            onClick={[Function]}
          />,
        ],
        "className": "slide-selector__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "className": "slide-selector__item slide-selector__item-active",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "className": "slide-selector__item ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
